### PR TITLE
Calculate test coverage in AutoRestReportTest.swift

### PR DESCRIPTION
**Pull request by sacheu**
* Rename AutoRestReportTest.swift to ZZZAutoRestReportTest.swift so that it will be the last test to run
* From the report and optional report, calculate the coverage based on Mobile tests and all tests
* Print out the result to console

### CODE REVIEW

File: AutorestSwiftTest/ZZZAutoRestReportTest.swift [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        // custom-baseUrl.json
+        "CustomBase",
+        // xms-error-response.json
+        "expected", "animalNotFoundError", "linkNotFoundError", "stringError", "intError",
+        // body-integer.json
+        "putInteger", "putLong", "putUnixTime", "getInteger", "getLong", "getUnixTime", "getInvalidUnixTime",
+        "getNullUnixTime",
+        // url.json
+        "UrlPathItem", "UrlQueries", "UrlPaths"
+    ]
```
Comment (tjprescott) [inline AutorestSwiftTest/ZZZAutoRestReportTest.swift:51]: As we add new testserver swaggers, do we need to update this (I'm assuming yes).

**Review by tjprescott — COMMENTED** [review]
Needs rebase off master and one question. Could you post an example of what the output looks like?

File: AutorestSwiftTest/ZZZAutoRestReportTest.swift (hunk 1 above)
Comment (sacheu) [reply AutorestSwiftTest/ZZZAutoRestReportTest.swift:51]: If the new test server swaggers are considered mobile scenarios, yes, we need to update this list. This list is gotten from the test server source code by looking at each endpoint and what is the name in the coverage report. that it built.

**Comment by sacheu** [comment]
See a regression in this PR branch after merge with master.

**Comment by sacheu** [comment]
Sample output:

Optional Coverage:
Failed mobile tests
-------------------
MultiapiCustomBaseUrlApiVersionTwo
MultiapiCustomBaseUrlApiVersionOne
Mobile Passed Test=0 Mobile Test=2 Coverage=0.0

Other passed tests
-------------------
HeaderParameterProtectedKey

Pass Test=1 Total Test=41 Coverage=0.024390243
Coverage:
Passed mobile tests
-------------------
expectedPetHungryError
getNullUnixTime
expectedNoErrors
UrlPathsBoolFalse
UrlPathsFloatNegative
putUnixTime
UrlQueriesDoubleNull
getLongOverflow
UrlQueriesDateValid
UrlQueriesDateTimeValid
expectedPetSadError
UrlQueriesBoolFalse
UrlQueriesDoubleNegative
UrlPathsEnumValid
putIntegerMin
intError
UrlPathItemGetGlobalNull
UrlPathsLongPositive
UrlPathsDoublePositive
UrlQueriesEnumNull
UrlPathsByteMultiByte
UrlQueriesLongPositive
UrlPathItemGetGlobalAndLocalNull
UrlPathsIntNegative
UrlQueriesBoolTrue
UrlQueriesByteEmpty
getIntegerOverflow
putLongMin
UrlQueriesLongNegative
getUnixTime
getInvalidUnixTime
UrlQueriesFloatNull
getIntegerUnderflow
getIntegerInvalid
animalNotFoundError
putIntegerMax
getIntegerNull
UrlQueriesDateTimeNull
UrlPathsIntPositive
UrlPathItemGetAll
UrlQueriesByteMultiByte
UrlPathsDateTimeValid
UrlPathsByteEmpty
UrlQueriesFloatNegative
linkNotFoundError
stringError
UrlPathsDateValid
UrlPathsLongNegative
UrlQueriesDateNull
UrlQueriesIntPositive
UrlQueriesEnumValid
UrlPathItemGetPathItemAndLocalNull
UrlQueriesIntNegative
getLongUnderflow
UrlPathsBoolTrue
UrlQueriesDoublePositive
UrlPathsDoubleNegative
putLongMax
Failed mobile tests
-------------------
unexpectedEnum
UrlPathsIntUnixTime
UrlQueriesBoolNull
UrlPathsArrayCSVInPath
UrlQueriesArrayCsvEmpty
UrlQueriesStringUnicode
UrlQueriesByteNull
UrlQueriesArrayCsvValid
CustomBaseUri
UrlQueriesFloatPositive
UrlQueriesArrayMultiNull
UrlQueriesStringUrlEncoded
UrlPathsStringUnicode
CustomBaseUriMoreOptions
UrlPathsStringUrlEncoded
UrlQueriesArrayCsvNull
UrlQueriesArrayPipesValid
expectedEnum
UrlQueriesArrayNoCollectionFormatValid
UrlPathsStringEmpty
UrlQueriesLongNull
UrlQueriesArrayMultiEmpty
UrlPathsFloatPositive
UrlPathsStringUrlNonEncoded
UrlQueriesArrayMultiValid
UrlQueriesStringEmpty
UrlQueriesIntNull
UrlPathsStringBase64Url
putModelFlattenCustomBase
UrlQueriesArrayTsvValid
UrlQueriesArraySsvValid
UrlQueriesStringNull
Mobile Passed Test=58 Mobile Test=90 Coverage=0.64444447

Other passed tests
-------------------
getModelFlattenResourceCollection
getModelFlattenDictionary
getModelFlattenArray
FileStreamNonempty
FileStreamEmpty
HttpSuccess200Head
FileStreamVeryLarge
HttpSuccess204Head
HttpSuccess404Head
putModelFlattenArray

Pass Test=68 Total Test=598 Coverage=0.11371238

**Comment by sacheu** [comment]
Merge issue is resolved.

**Comment by tjprescott** [comment]
> Other passed tests
> getModelFlattenResourceCollection
> getModelFlattenDictionary
> getModelFlattenArray
> FileStreamNonempty
> FileStreamEmpty
> HttpSuccess200Head
> FileStreamVeryLarge
> HttpSuccess204Head
> HttpSuccess404Head
> putModelFlattenArray
> 
> Pass Test=68 Total Test=598 Coverage=0.11371238

Two things. 

1) I would print the coverage values as a %.

2) Is a log generated for this? If so, I might suggest printing the individual pass/failed tests to the log and keeping the console output more succinct.

**Comment by sacheu** [comment]
The mobile relevant test cases are 90 out of 598. And we pass 58 mobile relevant test cases.
So our coverage is 64%

**Comment by tjprescott** [comment]
> The mobile relevant test cases are 90 out of 598. And we pass 58 mobile relevant test cases.
> So our coverage is 64%

In that case, I think you should change the algorithm so that it accurately displays that. Right now it says 11%.

Also, if it's easy to do, it would be nice, for the mobile-relevant swaggers, to display the stats for that specific test swagger (i.e. 10/10 (100%) test cases)

That way we communicate which swaggers are DONE and which are in progress.

**Comment by sacheu** [comment]
The report does have

Mobile Passed Test=58 Mobile Test=90 Coverage=0.64444447

but it is not obvious since i am also listing out what are the passing test cases and failed ones. I am working on moving them to a log file. Having comiling issues due to new dependencies.

**Review by tjprescott — APPROVED** [review]

**Pull request merged**